### DISCUSSION TITLE
GND calculation for hcp materials

### DISCUSSION THREAD

**Discussion by DanSalehiUT** [mtex-toolbox/mtex#1842] [Ask Anything]
Hi,

I'm trying to calculate GND for Mg AZ31B alloy which has a hcp crystal structure, but the problem is that the dislocation systems are very well defined for BCC and FCC but not for hcp systems. How can I find the total GND values as well as GND values for basal, pyramidal, prismatic slips? Any help would be appreciated!

Thanks,

Danial

**Comment by lfgmorales** [comment]
Hi

I am not entirely sure I understood the question, but the slip systems for hcp are also well know and even pre-defined on mtex, have a look here

<https://mtex-toolbox.github.io/TaylorHex.html>
Texture evolution in rolled magnesium during uniaxial tension | MTEX<https://mtex-toolbox.github.io/TaylorHex.html>
mtex-toolbox.github.io<https://mtex-toolbox.github.io/TaylorHex.html>
        [favicon.png] <https://mtex-toolbox.github.io/TaylorHex.html>


From there you can combine with GND

Best

Luiz

Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 26 May 2023, at 23:32, DanSalehiUT ***@***.***> wrote:

﻿

Hi,

I'm trying to calculate GND for Mg AZ31B alloy which has a hcp crystal structure, but the problem is that the dislocation systems are very well defined for BCC and FCC but not for hcp systems. How can I find the total GND values as well as GND values for basal, pyramidal, prismatic slips? Any help would be appreciated!

Thanks,

Danial

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1842>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN3R3DGWN3XZCQ6PLZ3XIEON3ANCNFSM6AAAAAAYQWWJW4>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1842", "url": "https://github.com/mtex-toolbox/mtex/discussions/1842", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Comment by DanSalehiUT** [comment]
Thanks for your response, Luiz! 

If the assumption of equal dislocation energy would be correct. I think the following code should work, but it did not converge. Have any ideas where I might have made a mistake?

sShcp = [slipSystem.basal(CS,1),...
  slipSystem.prismatic2A(CS,66),...
  slipSystem.pyramidalCA(CS,80)];
[sScold,slipId] = sShcp.symmetrise;
dS =  dislocationSystem(sShcp);
% energy of the edge dislocations
dS(dS.isEdge).u = 1;

% energy of the screw dislocations
dS(dS.isScrew).u = 1 - 0.3;
dSRot = ebsd2.orientations * dS;
[rho,factor] = fitDislocationSystems(kappa,dSRot);
